### ISSUE TITLE
Running the 'idea' task in buildSrc causes NullPointerException

### ISSUE THREAD

**Issue by atrocities** [gradle/gradle#5110]
Starting in Gradle 4.6 (and in 4.7), applying the 'idea' plugin and running the 'idea' task in buildSrc no longer works.

### Expected Behavior
Applying the 'idea' plugin to buildSrc and adding the 'idea' task as a dependency of 'build' in buildSrc should create IntelliJ project files, allowing buildSrc to be opened in IDEA.

### Current Behavior
Running the IDEA task in buildSrc causes an NPE
```
Could not determine the dependencies of task ':buildSrc:idea'.                                                                                                                                          [33/1320]
> java.lang.NullPointerException (no error message)

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.internal.tasks.TaskDependencyResolveException: Could not determine the dependencies of task ':buildSrc:idea'.
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:62)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.addToTaskGraph(DefaultTaskExecutionPlan.java:179)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.addTasks(DefaultTaskGraphExecuter.java:113)
        at org.gradle.execution.TaskNameResolvingBuildConfigurationAction.configure(TaskNameResolvingBuildConfigurationAction.java:47)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:48)
        at 
...

Caused by: java.lang.NullPointerException
        at org.gradle.composite.internal.DefaultProjectPathRegistry$ProjectPathEntry.access$100(DefaultProjectPathRegistry.java:78)
        at org.gradle.composite.internal.DefaultProjectPathRegistry.getProjectComponentIdentifier(DefaultProjectPathRegistry.java:70)
        at org.gradle.plugins.ide.idea.IdeaPlugin.allImlArtifactsInComposite(IdeaPlugin.java:505)
        at org.gradle.plugins.ide.idea.IdeaPlugin.access$900(IdeaPlugin.java:77)
        at org.gradle.plugins.ide.idea.IdeaPlugin$22.call(IdeaPlugin.java:497)
        at org.gradle.plugins.ide.idea.IdeaPlugin$22.call(IdeaPlugin.java:494)
        at org.gradle.util.GUtil.uncheckedCall(GUtil.java:458)
        at org.gradle.api.internal.tasks.DefaultTaskDependency.visitDependencies(DefaultTaskDependency.java:122)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:89)
        at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmpyEdges.getNodeValues(CachingDirectedGraphWalker.java:211)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:121)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:73)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.doGetDependencies(CachingTaskDependencyResolveContext.java:76)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:60)
        ... 105 more
```

### Context
With a large amount of code in buildSrc, it's important to have this code open-able in an IDE.

A quick workaround is to cram the following into the buildSrc/build.gradle.
`project.getServices().get(ProjectPathRegistry.class).add(project.getIdentityPath(), ":buildSrc", new DefaultProjectComponentIdentifier(new DefaultBuildIdentifier(":buildSrc"), ":buildSrc"), false)`

However, this is obviously super brittle since it touches a lot of internal stuff in Gradle.

### Steps to Reproduce (for bugs)
- Create a buildSrc directory in any Gradle 4.6 or 4.7 project
- Add a build.gradle file in buildSrc/ with contents
```
apply plugin: 'idea'

tasks.build.dependsOn tasks.idea
```
- Run `./gradlew tasks --stacktrace`
- Observe NPE

### Your Environment
MacOS High Sierra, Java 8u162

**Comment by oehme** [comment]
@adammurdoch I remember you did some changes in that area, could you take a look at this one?

**Comment by adammurdoch** [comment]
This is fixed in 4.8.
